apps/aecore/src/aec_state.erl

```diff
@@ -124,25 +125,34 @@ code_change(_OldVsn, State, _Extra) ->
 
 %% INFO: not optimized (by local lookup of prev-block by hash in current-block)
 %%       because of incoming optimization with check points
--spec setup_trees(height(), trees()) -> {ok, trees()}.
+-spec setup_trees(non_neg_integer(), trees()) -> {ok, trees()}.
 setup_trees(0, Trees) ->
     {ok, Trees};
 setup_trees(N, Trees) ->
     do_setup_trees(0, N, Trees).
 
--spec do_setup_trees(height(), height(), trees()) -> {ok, trees()}.
+-spec do_setup_trees(non_neg_integer(), non_neg_integer(), trees()) -> {ok, trees()} | {error, term()}.
 do_setup_trees(TopHeight, TopHeight, Trees) ->
     {ok, Trees};
 do_setup_trees(CurrentHeight, TopHeight, TreesAtEndOfPreviousBlock) ->
-    {ok, Block} = aec_chain:get_block_by_height(CurrentHeight),
-    Txs = aec_blocks:txs(Block),
-    {ok, TreesUpdated} = apply_txs_internal(Txs, CurrentHeight, TreesAtEndOfPreviousBlock),
-    do_setup_trees(CurrentHeight+1, TopHeight, TreesUpdated).
+    case aec_chain:get_block_by_height(CurrentHeight) of
+        {ok, Block} ->
+            Txs = aec_blocks:txs(Block),
+            case apply_txs_internal(Txs, CurrentHeight, TreesAtEndOfPreviousBlock) of
+                {ok, TreesUpdated} ->
+                    do_setup_trees(CurrentHeight+1, TopHeight, TreesUpdated);
+                Error ->
```

Comment:
May I suggest avoiding this open ending case? It help maintainability. E.g. `{error, _} = Error ->` is better.